src/js/components/Select/Select.js

```diff
@@ -186,11 +186,26 @@ const Select = forwardRef(
         setValue(nextValue);
         if (onChange) {
           event.persist();
-          const adjustedEvent = event;
-          adjustedEvent.target = inputRef.current;
-          adjustedEvent.value = nextValue;
-          adjustedEvent.option = option;
-          adjustedEvent.selected = nextSelected;
+          const adjustedEvent =
+            event instanceof Event
+              ? // native event
+                new event.constructor(event.type, event)
+              : // react SyntheticBaseEvent
+                new event.constructor(
```

Comment:
Well, I didn't realize it would be this involved. This feels like it's too intimate with the internals of React. I think the way you had it before is preferable. Thanks for trying this out though. Sorry for the flip-flop.

Reply:
No problem. I had the same idea, but gave up when I saw that it looks complicated.
I reverted it.